cmd/clusterctl/client/config_test.go

```diff
@@ -81,30 +84,26 @@ func Test_clusterctlClient_GetProvidersConfig(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := tt.field.client.GetProvidersConfig()
-			if tt.wantErr != (err != nil) {
-				t.Fatalf("error = %v, wantErr = %v", err, tt.wantErr)
-			}
 			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
 				return
 			}
 
-			if len(got) != len(tt.wantProviders) {
-				t.Errorf("got = %v items, want %v items", len(got), len(tt.wantProviders))
-				return
-			}
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(got).To(HaveLen(len(tt.wantProviders)))
 
-			for i, g := range got {
+			for i, gotProvider := range got {
```

Comment:
I think this loop can likely be replaced with `g.Expect(got).To(ConsistOf(tt.wantProviders))`

Reply:
the `tt.wantProviders` contains only the name of the providers and the `got` the full object like
```

{
                    name: "vsphere",
                    url: "https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/releases/latest/infrastructure-components.yaml",
                    providerType: "InfrastructureProvider",
                },
```

this test fail if use `g.Expect(got).To(ConsistOf(tt.wantProviders))`

Reply:
Ah, indeed...